backups/README.md

```diff
@@ -0,0 +1,198 @@
+Google Cloud Storage (GCS). They support the commit-edge architecture and will backup
+individual checkpoints.
+
+## Prerequisites
+
+This guide assumes you have a Helix Core server (or servers, in the commit-edge architecture)
+running on Linux. It furthermore assumes the p4d instances run on the Google Cloud Platform,
+although the scripts can be easily adjusted to run elsewhere.
+
+Additionally, you will need a Google Cloud account with billing enabled to use GCS.
+
+## Strategy
+
+The scripts in this directory implement a backup strategy that consists of using two GCS buckets:
+
+### 1. Incremental
+
+It is used for a nighlty rsync operation that copies only the versioned files that changed.
+
+This bucket allows both writes and deletes because Perforce will sometimes replace the contents
+of versioned files and/or remove them. This is essentially a live mirror of the versioned data
+and should update reasonably fast. However, it offers a higher attack surface since a bad actor
+could theoretically delete files if they gained access to the backup service account.
+
+### 2. Regular
+
+It is used for a nighlty checkpoint backup and a full monthly versioned files backup.
+
+This bucket only allows write operations, meaning that a malicious user would have harder time
+deleting the files. However, the full backup operation takes a much longer time than an incremental
+rsync.
+
+### Note on retention
+
+GCS allows to configure file rentention per buckets. It is recommended to set indefinite retention
+on the Incremental bucket and a 90-day retention for the Regular bucket.
+
+However, these parameters (as well as the backup cadence) can be adjusted based on the needs.
+
+### Offline root
+
+The nightly backup screen uses an offline copy of the Helix Core database to avoid locking the
+server during the checkpoint operations. Please make sure that it has been created and seeded with
+from a main server checkpoint.
+
+## Setup
+
+## GCS Buckets
+
+### Create buckets
+
+Create the buckets by navigating to Storage | Browser in the Google Cloud Console and
+select "Create Bucket":
+
+![Create Storage Bucket](create_storage_bucket.png)
+
+Bucket names must be globally unique. One easy way of addressing this requirement is to use
+the project name as a suffix. For example, if your project is called my-perforce-123, you
+can name the Incremental bucket *incremental-my-perforce-123*.
+
+There are a number of considerations for choosing the location type and storage class.
+We recommend reviewing Google Cloud documentation to understand the tradeoffs involved in
+making those choices. For example, a Multi-region location type would incur the highest cost
+but will provide the highest availability. A Nearline storage class could be appropriate for
+backups, but you would need to evaluate this based on cost and performance characteristics.
+
+We recommend configuring a uniform access to all objects in the bucket.
+
```

Comment:
Recommend adding a note about verifying that bucket is not publicly readable, due to sensitive data.

Reply:
Added a note.